### PR TITLE
support for multiple instances for memcached

### PR DESCRIPTION
**Pull request by mmckinst**
This also adds support for memcached via sockets.

### CODE REVIEW

**Comment by mmckinst** [comment]
FWIW, the failing build is unrelated to this pull request, the [current master build is failing right now too](https://travis-ci.org/voxpupuli/puppet-collectd/builds/139840763).

**Comment by mmckinst** [comment]
Related to the [2.0.0 release of puppet-lint yesterday](https://github.com/rodjek/puppet-lint/commit/889f21fc03a56867e4fe4925b24b3eab910977e8) whereas puppet-lint-variable_contains_upcase requires puppet-lint < 2.0.  There is a [pull request to fix it](https://github.com/fiddyspence/puppetlint-variablecase/pull/2).

**Comment by mmckinst** [comment]
Documentation for instances and sockets [here](https://github.com/collectd/collectd/blob/7b64cc91d3307179557b25b6ece2349a088f4294/src/collectd.conf.pod#plugin-memcached).

File: README.md [hunk 1]
```diff
@@ -821,10 +821,26 @@ class { 'collectd::plugin::lvm': }
...
+
 ```puppet
 class { 'collectd::plugin::memcached':
-  host => '192.168.122.1',
-  port => 11211,
+  instances => {
+    'sessions1' => {
+      'host' => '127.0.0.1',
+	  'port' => '11211',
+	}
```
Comment (ffrank) [inline README.md]: The indentation is a little chaotic. Could you clean that up please?

File: examples/plugins/memcached.pp [hunk 2]
```diff
@@ -1,7 +1,11 @@
 include ::collectd
 
 class { '::collectd::plugin::memcached':
-  host => '192.168.122.1',
-  port => '11211',
+  instances => {
+    'default' => {
+      'host' => '127.0.0.1',
```
Comment (ffrank) [inline examples/plugins/memcached.pp]: Is this a more sensible default that the old `192.168.122.1`? (I suppose it doesn't really matter.)

**Comment by mmckinst** [comment]
@ffrank I've fixed the issues you mentioned. Its now failing on linting, claiming there are no trailing commas when there clearly are. I'll look at it tomorrow to see why its failing.

**Comment by dhoppe** [comment]
@mmckinst This happens to me all the time. Maybe I should add `puppet-lint -f ...` and `rubocop -a ...` as Git pre-commit hook. 

In this case `puppet-lint` is expecting:

```
include ::collectd

class { '::collectd::plugin::memcached':
  instances => {
    'default' => {
      'host' => '127.0.0.1',
      'port' => '11211',
    },
  },
}
```

**Comment by mmckinst** [comment]
Everything is passing now!

**Comment by ffrank** [comment]
Cool. My comments from earlier are now obsolete.

Last nit: The syntax and formatting in the README example are off. However, we can go ahead and merge this as is. The README can be fixed later.

LGTM, will someone push the button? @dhoppe?

**Comment by mmckinst** [comment]
@ffrank README should be fixed now

**Pull request merged**